src/bp-members/classes/class-bp-members-notices-admin.php

```diff
@@ -0,0 +1,405 @@
+					<div class="form-input">
+						<input type="url" value="<?php echo esc_url( $form_values['url'] ); ?>" class="bp-panel-input regular-text code" id="bp_notice_url" name="bp_notice[url]" size="40" aria-describedby="bp-url-description" />
+						<p id="bp-url-description"><?php esc_html_e( 'The action button url to head user to.', 'buddypress' ); ?></p>
+					</div>
+				</div>
+				<div class="form-field">
+					<label for="bp_notice_text"><?php esc_html_e( 'Action button text', 'buddypress' ); ?></label>
+					<div class="form-input">
+						<input type="text" value="<?php echo esc_attr( $form_values['text'] ); ?>" class="bp-panel-input regular-text code" id="bp_notice_text" name="bp_notice[text]" size="40" aria-describedby="bp-text-description" />
+						<p id="bp-text-description"><?php esc_html_e( 'The text of the action button.', 'buddypress' ); ?></p>
+					</div>
+				</div>
+				<p class="submit">
+					<?php if ( ! $is_edit ) : ?>
+						<input type="submit" value="<?php esc_attr_e( 'Publish Notice', 'buddypress' ); ?>" name="bp_notice[send]" class="button button-primary save alignleft">
+					<?php else : ?>
+						<input type="hidden" value="<?php echo esc_attr( $form_values['id'] ); ?>" name="bp_notice[id]" />
+						<input type="submit" value="<?php esc_attr_e( 'Update Notice', 'buddypress' ); ?>" name="bp_notice[update]" class="button button-primary save alignleft">
+					<?php endif; ?>
+					<span class="spinner"></span>
+				</p>
+			</form>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Generate content to manage all notices.
+	 *
+	 * @since 15.0.0
+	 */
+	public function manage_notices() {
+		$this->list_table->prepare_items();
+		?>
+		<div class="wrap nosubsub">
+			<h1 class="wp-heading-inline"><?php echo esc_html_x( 'Member Notices', 'Notices admin page title', 'buddypress' ); ?></h1>
+			<hr class="wp-header-end">
+
+			<?php if ( isset( $_GET['bp-success'] ) || isset( $_GET['bp-error'] ) ) : ?>
+				<div id="message" class="<?php echo isset( $_GET['bp-success'] ) ? 'updated' : 'error'; ?> notice is-dismissible">
+					<p>
+						<?php
+						if ( isset( $_GET['bp-error'] ) ) {
+							if ( 'create' === $_GET['bp-error'] ) {
+								esc_html_e( 'Notice was not created. Please try again.', 'buddypress' );
+							} else {
+								esc_html_e( 'Notice was not updated. Please try again.', 'buddypress' );
+							}
+						} else {
+							if ( 'create' === $_GET['bp-success'] ) {
+								esc_html_e( 'Notice successfully created.', 'buddypress' );
+							} else {
+								esc_html_e( 'Notice successfully updated.', 'buddypress' );
+							}
+						}
+						?>
+					</p>
+				</div>
+			<?php endif; ?>
+
+			<div id="col-container" class="wp-clearfix">
+				<div id="col-left">
+					<div class="col-wrap">
+						<?php $this->notice_form(); ?>
+					</div>
+				</div><!-- /col-left -->
+				<div id="col-right">
+					<div class="col-wrap">
+						<?php $this->list_table->display(); ?>
```

Comment:
I think one possible feature request we might get is to show the total number of users that read/dismissed the notice.

![CleanShot 2024-09-02 at 13 16 09@2x](https://github.com/user-attachments/assets/3d8f4889-f797-47af-be07-0d835a4779ce)

Reply:
I agree, I thought about it myself 👍 . I'll see what I can do about it once most of the PR has been committed 😉